### PR TITLE
Enable SqlDataReader.GetValue() and GetSqlValue() for UDT types.

### PR DESCRIPTION
**Pull request by corivera**
Currently the UDT data is just returned as bytes.

### CODE REVIEW

File: src/System.Data.SqlClient/tests/ManualTests/SQL/UdtTest/UdtTest.cs [hunk 1]
```diff
@@ -40,6 +40,28 @@ public static void GetSchemaTableTest()
         }
 
         [CheckConnStrSetupFact]
+        public static void GetValueTest()
+        {
+            using (SqlConnection conn = new SqlConnection(DataTestUtility.TcpConnStr))
```
Comment (geleems) [inline src/System.Data.SqlClient/tests/ManualTests/SQL/UdtTest/UdtTest.cs]: `using` statement does not need block? I don't see opening and ending brackets.

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDataReader.cs [hunk 2]
```diff
@@ -2350,23 +2350,7 @@ private object GetSqlValueFromSqlBufferInternal(SqlBuffer data, _SqlMetaData met
             else if (_typeSystem != SqlConnectionString.TypeSystem.SQLServer2000)
             {
                 // TypeSystem.SQLServer2005
```
Comment (geleems) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDataReader.cs]: Can you change to the comment to ``// TypeSystem.SQLServer2005 and above`` like other spots?

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDataReader.cs [hunk 3]
```diff
@@ -2536,23 +2520,7 @@ private object GetValueFromSqlBufferInternal(SqlBuffer data, _SqlMetaData metaDa
...
```
Comment (geleems) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDataReader.cs]: Also, can you change to the comment to ``// TypeSystem.SQLServer2005 and above`` as well?

File: src/System.Data.SqlClient/tests/ManualTests/SQL/UdtTest/UdtTest.cs (hunk 1 above)
Comment (corivera) [reply src/System.Data.SqlClient/tests/ManualTests/SQL/UdtTest/UdtTest.cs]: You can chain them together like this to avoid using lots of curly braces.

**Comment by geleems** [comment]
@dotnet-bot test Tizen armel Debug Build please

**Comment by geleems** [comment]
@dotnet-bot test Linux x64 Release Build please

**Comment by corivera** [comment]
@dotnet-bot Test Linux x64 Release Build please

**Comment by corivera** [comment]
@dotnet-bot Test Linux x64 Release Build please

**Pull request merged**